server/controllers/documents.js

```diff
@@ -0,0 +1,289 @@
+import helper from '../helpers/helper';
+import models from '../models/index';
+
+/**
+   * Creates a new instance of document in the database
+   * @function createDocument
+   * @param {object} request request
+   * @param {object} response response
+   * @return {object} object - information about the request
+   */
+const createDocument = (request, response) => {
+  helper.verifyDocumentParams(request)
+  .then((result) => {
+    const verifiedParams = result.mapped();
+    const noErrors = result.isEmpty();
+    if (!noErrors) {
+      return response.status(412).json({
+        message: verifiedParams,
+        more_info: 'https://dmsys.herokuapp.com/#creates-new-documents'
+      });
+    }
+    models.Document.create({
+      title: request.body.title,
+      content: request.body.content,
+      access: request.body.access,
+      userId: request.decoded.data.userId,
+      roleId: request.decoded.data.roleId,
+    })
+    .then(document => response.status(201).json({
+      message: 'Document created',
+      title: document.title,
+      ownerId: request.decoded.data.userId,
+    })
+    )
+    .catch(error => response.status(409).json({
+      message: 'An unexpected error occurred',
+      error: error.errors,
+      more_info: 'https://dmsys.herokuapp.com/#creates-new-documents'
+    })
+    );
+  });
+};
+
+
+/**
+   * fetches all documents in the database according to the users' access right
+   * @function getAllDocument
+   * @param {object} request request
+   * @param {object} response response
+   * @return {object} object - nformation about the documents
+   * in the database
+   */
+const getAllDocument = (request, response) => {
+  const query = helper.queryForAllDocuments(request);
+  models.Document.findAndCountAll(query)
+  .then((documents) => {
+    if (documents.count < 1) {
+      return response.status(404).json({
+        message: 'Access Denied',
```

Comment:
I don't think Access denied is a good message to return when no document is returned